File: src/main/java/dev/iancmd/operationrelife/init/ModItems.java
package dev.iancmd.operationrelife.init;

import dev.iancmd.operationrelife.core.OperationRelife;
import net.minecraft.world.item.BlockItem;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.Rarity;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import net.minecraftforge.registries.RegistryObject;

public class ModItems {
  public static DeferredRegister<Item> getRegister() {
    return ModItems.ItemRegistry.REGISTER;
  }
  public static class ItemRegistry {
    public static final DeferredRegister<Item> REGISTER = DeferredRegister.create(ForgeRegistries.ITEMS, OperationRelife.MODID);
    public static final RegistryObject<Item> TEST_BLOCK = ItemRegistry.REGISTER.register("test_block", () -> new BlockItem(ModBlocks.BlockRegistry.TEST_BLOCK.get(), new Item.Properties().rarity(Rarity.EPIC)));
    public static final RegistryObject<Item> RED_SOUL_HOLDER = ItemRegistry.REGISTER.register("red_soul_holder", () -> new BlockItem(ModBlocks.BlockRegistry.RED_SOUL_HOLDER.get(), new Item.Properties().rarity(Rarity.COMMON)));
    public static final RegistryObject<Item> BLUE_SOUL_HOLDER = ItemRegistry.REGISTER.register("blue_soul_holder", () -> new BlockItem(ModBlocks.BlockRegistry.BLUE_SOUL_HOLDER.get(), new Item.Properties().rarity(Rarity.COMMON)));
  }
}

File: src/main/kotlin/dev/iancmd/operationrelife/init/ModItems.kt
package dev.iancmd.operationrelife.init

import dev.iancmd.operationrelife.core.OperationRelife
import dev.iancmd.operationrelife.init.ModBlocks.BlockRegistry
import dev.iancmd.operationrelife.init.ModItems.ItemRegistry.REGISTER
import net.minecraft.world.item.BlockItem
import net.minecraft.world.item.BucketItem
import net.minecraft.world.item.Item
import net.minecraft.world.item.Rarity
import net.minecraftforge.registries.DeferredRegister
import net.minecraftforge.registries.ForgeRegistries
import net.minecraftforge.registries.RegistryObject

object ModItems {
  fun getRegister(): DeferredRegister<Item> {
    return REGISTER
  }

  object ItemRegistry {
    private val register: DeferredRegister<Item> =
      DeferredRegister.create(ForgeRegistries.ITEMS, OperationRelife.Companion.MODID)
    val REGISTER: DeferredRegister<Item>
      get() = register

    val RED_SOUL_HOLDER: RegistryObject<Item> = register.register<Item>("red_soul_holder") {
      BlockItem(
        BlockRegistry.RED_SOUL_HOLDER.get(), Item.Properties().rarity(
          Rarity.COMMON
        )
      )
    }
    val BLUE_SOUL_HOLDER: RegistryObject<Item> = register.register<Item>("blue_soul_holder") {
      BlockItem(
        BlockRegistry.BLUE_SOUL_HOLDER.get(), Item.Properties().rarity(
          Rarity.COMMON
        )
      )
    }
    val UPPER_SURFACE_ICE: RegistryObject<Item> = register.register<Item>("upper_surface_ice") {
      BlockItem(
        BlockRegistry.UPPER_SURFACE_ICE.get(), Item.Properties().rarity(
          Rarity.COMMON
        )
      )
    }
    val COLD_FLUID_BUCKET: RegistryObject<Item> = register.register<Item>("cold_fluid_bucket") {
      BucketItem(
        ModFluids.FluidRegistry.COLD_FLUID_SOURCE, Item.Properties()
      )
    }
  }
}
